crates/nu-command/src/filesystem/rm.rs

```diff
@@ -183,11 +188,25 @@ fn rm(
 
     let span = call.head;
 
+    #[cfg(all(
+        feature = "trash-support",
+        not(target_os = "android"),
+        not(target_os = "ios")
+    ))]
     let config = engine_state.get_config();
 
+    #[cfg(all(
+        feature = "trash-support",
+        not(target_os = "android"),
+        not(target_os = "ios")
+    ))]
     let rm_always_trash = config.rm_always_trash;
 
-    #[cfg(not(feature = "trash-support"))]
+    #[cfg(all(
+        feature = "trash-support",
```

Comment:
I think your code actually inverted this condition so that the error on line 224 ("Cannot execute `rm` with option `--trash`; feature `trash-support` not enabled") does not make sense because `trash-support` is enabled.

Reply:
Ah oops you're right, this is a copy paste error. I just did this to fix compile warnings about unused variables. I'll just leave this out of this PR and open a separate one to address it, since it's a bit orthogonal and a nontrivial to reason about.

Reply:
Yes, opening new PRs for separate changes is very much preferred.